gcsa/google_calendar.py

```diff
@@ -40,20 +41,24 @@ def __init__(self,
                 if require read only access. Default: False
         :param application_name:
                 name of the application. Default: None
+        :param token_path:
+                path to save authenticated token file. Default: "token.pickle"
         """
         credentials_path = credentials_path or _get_default_credentials_path()
         self._credentials_dir, self._credentials_file = os.path.split(credentials_path)
 
         self._scopes = [self._READ_WRITE_SCOPES + ('.readonly' if read_only else '')]
         self._application_name = application_name
+        self._token_path = token_path
 
         self.calendar = calendar
         credentials = self._get_credentials()
         self.service = build('calendar', 'v3', credentials=credentials)
 
     def _get_credentials(self):
         _credentials_path = os.path.join(self._credentials_dir, self._credentials_file)
-        _token_path = os.path.join(self._credentials_dir, 'token.pickle')
+        _default_token_path = os.path.join(self._credentials_dir, 'token.pickle')
+        _token_path = self._token_path or _default_token_path
```

Comment:
I'd move this logic to `__init__` so we don't keep `self._token_path` uninitialized. And just use it here straight away.

Reply:
Just to make sure I understand;

initialize `self._token_path` in `__init__` like `        self._token_path = _token_path or os.path.join(self._credentials_dir, 'token.pickle')` and, replace all the usages of `_token_path` in `_get_credentials` with `self._token_path`?

Reply:
Yes, exactly